internal/pkg/consensus/validation.go

```diff
@@ -47,54 +47,41 @@ func init() {
 	errNonceTooHighCt = metrics.NewInt64Counter("consensus/msg_nonce_high_err", "Number of messages with nonce too high")
 }
 
-// DefaultMessageValidator validates incoming signed messages.
-type DefaultMessageValidator struct {
-	allowHighNonce bool
+// MessageSelectionChecker checks for miner penalties on signed messages
+type MessagePenaltyChecker struct {
+	api penaltyCheckerAPI
 }
 
-// NewDefaultMessageValidator creates a new default validator.
-// A default validator checks for both permanent semantic problems (e.g. invalid signature)
-// as well as temporary conditions which may change (e.g. actor can't cover gas limit).
-func NewDefaultMessageValidator() *DefaultMessageValidator {
-	return &DefaultMessageValidator{}
+// penaltyCheckerAPI allows the validator to access latest state
+type penaltyCheckerAPI interface {
+	Head() block.TipSetKey
+	GetActorAt(ctx context.Context, tipKey block.TipSetKey, addr address.Address) (*actor.Actor, error)
 }
 
-// NewOutboundMessageValidator creates a new default validator for outbound messages. This
-// validator matches the default behaviour but allows nonces higher than the actor's current nonce
-// (allowing multiple messages to enter the mpool at once).
-func NewOutboundMessageValidator() *DefaultMessageValidator {
```

Comment:
I've done away with this abstraction because we're not doing *any* high nonce validation on messages that we put in the mpool after this PR.  I am probably least sure about the ramifications here so please take a look and let me know if this is something I need to think more about.

Reply:
We're going to need something here. "Validation" isn't the right term, but we need something to limit mpool spam, and evict lower-likelihood messages before better-looking ones. An existant sender and a limit on the number of distinct nonces from each sender will be important parts of that.

Needn't block this PR, but it's an important thing to get back. Please file an issue.

Reply:
#3872